### PR TITLE
Make default selected check category filterable

### PR DESCRIPTION
**Pull request by ernilambar**
Related https://github.com/WordPress/plugin-check/issues/494

* Introduce filter `wp_plugin_check_default_categories`
* Add new method `get_default_check_categories_to_be_selected()` in `Admin_Page` class
* Add test for related changes

### CODE REVIEW

File: includes/Admin/Admin_Page.php [hunk 1]
```diff
@@ -300,4 +302,28 @@ public function admin_footer() {
...
 	}
+
+	/**
+	 * Gets default check categories to be selected.
+	 *
+	 * @since 1.0.2
+	 *
+	 * @return string Categories separated by double underscores.
+	 */
+	public function get_default_check_categories_to_be_selected() {
```
Comment (swissspidy) [inline includes/Admin/Admin_Page.php]: No need for this to be public

Comment (ernilambar) [reply includes/Admin/Admin_Page.php]: Yah, I understood. But when I kept private/protected I could not implement unit test for it as it gave fatal error. Any suggestions?

Comment (swissspidy) [reply includes/Admin/Admin_Page.php]: We could test it via `render_page()` (like the `test_render_page` tests)

File: includes/Admin/Admin_Page.php [hunk 2]
```diff
@@ -300,4 +302,28 @@ public function admin_footer() {
...
+		$default_check_categories = array(
+			'plugin_repo',
+		);
+
+		/**
+		 * Filters the default check categories to be selected.
+		 *
+		 * @since 1.0.2
+		 *
+		 * @param array $default_check_categories An array of categories.
```
Comment (swissspidy) [inline includes/Admin/Admin_Page.php]:
```suggestion
		 * @param string[] $default_check_categories An array of category slugs.
```

File: includes/Admin/Admin_Page.php [hunk 3]
```diff
@@ -300,4 +302,28 @@ public function admin_footer() {
...
+		 */
+		$default_categories = (array) apply_filters( 'wp_plugin_check_default_categories', $default_check_categories );
+
+		return ! empty( $default_categories ) ? implode( '__', $default_categories ) : '';
```
Comment (swissspidy) [inline includes/Admin/Admin_Page.php]: No need for the `empty` check. `implode` already returns an empty string when the array is empty.

Comment (swissspidy) [inline includes/Admin/Admin_Page.php]: I don't like doing the `implode` here. Let's do it above.

```php
$user_enabled_categories = get_user_setting( 'plugin_check_category_preferences', implode( '__', $this->get_default_check_categories_to_be_selected() ) );

**Review by swissspidy — APPROVED** [review]

**Review by mukeshpanchal27 — APPROVED** [review]

**Pull request merged**